web/src/main/java/org/springframework/security/web/access/DelegatingMissingAuthorityAccessDeniedHandler.java

```diff
@@ -247,17 +247,16 @@ public DelegatingMissingAuthorityAccessDeniedHandler build() {
 	 */
 	private static final class AuthorityRequiredFactorErrorEntry {
 
-		private final String authority;
+		@Nullable private final String authority;
 
 		private final @Nullable RequiredFactorError error;
 
-		private AuthorityRequiredFactorErrorEntry(String authority, @Nullable RequiredFactorError error) {
-			Assert.notNull(authority, "authority cannot be null");
+		private AuthorityRequiredFactorErrorEntry(@Nullable String authority, @Nullable RequiredFactorError error) {
```

Comment:
If both are null, then this object doesn't provide any error information. Let's continue to require that the authority be non-null (please see earlier comment).